www/perl-lib/SMT/Utils.pm

```diff
@@ -127,6 +129,46 @@ sub db_connect
     return $dbh;
 }
 
+=item hasRegSharing()
+
+Check whether registration sharing is enabled or not.
+
+=cut
+
+sub hasRegSharing
+{
+    my $r = shift;
+
+    if (! defined $REGSHARING) {
+        $REGSHARING = 0;
+        my $cfg;
+        eval
+        {
+            $cfg = SMT::Utils::getSMTConfig();
+        };
+        if($@ || !defined $cfg)
+        {
+            if (! $r)
+            {
+                return;
+            }
+            $r->log_error("Cannot read the SMT configuration file: ".$@);
+            my $msg = 'SMT server is missconfigured. Please contact your '
+                . 'administrator.';
+            return http_fail($r, 500, $msg);
```

Comment:
Hmm, I was following the established pattern, see for example: Registration.pm lines 166-171

I have no issue with the code change, but would like to understand why this case would be different than other cases where we check for a successful load of the configuration.